_assets/ci/isolate-vendor-check.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# This is a hack to isolate vendor check in a
+# separate clean state. Without this, validate-vendor.sh
+# doesn't like our workflow with patches.
+#
+# How it works:
+# 1) Stashes all changes and checks out to a temporary branch.
+# 2) Reverts all patches and commits changes.
+# 3) Runs "dep ensure" and validate-vendor.sh. Saves exit code and message.
+# 4) Commits any changes.
```

Comment:
About first commit: Running the validation script when there are patches, it says "Oh, you've made some changes! That's not the way to work with `dep`." Then it removes all our patches and complains *until we commit*, which is bad because now it removed all of the changes it's not possible to check any other changes. And this script is about checking those ones because ideally there should be no other changes than patches. After reversion and committing, what's left are those.

About second commit: It rarely leaves the workspace in an uncommitted state so that's useful before `git checkout <previous_branch>`.

Reply:
Yeah, I was just wondering what was the reason to not do `git reset --hard` instead.

Reply:
That was my first option but this looked like a cleaner approach than counting successful commits. And safer while making changes on the script.